fix: rename files during the transfer, adding the .partial extension

DeboraTahara opened the pull request.

DeboraTahara committed:
fix: rename files during the transfer, adding the .partial extension

DeboraTahara commented:
ping @doudou

doudou reviewed (commented):
Missing the work on log_runtime_archive to ignore the .partial files.

DeboraTahara committed:
fix: ignore .partial files during archiving

doudou committed:
fix: add support for renames in our internal FTP server for log transfer

doudou committed:
fix: error handling in FTPUpload

The errors were simply "eaten" by the code. Add test suite for FTPUpload

doudou committed:
feat: re-add .partial extension during upload

doudou commented:
@wvmcastro @kapeps I took this PR over ... could you have a look please :

doudou committed:
chore: cleanup the SpawnServer tests

doudou committed:
chore: cleanup LogRuntimeArchiveMain unit tests

The tests are full of pretend variables that are actually fixed values, and made
changing the tests very hard (e.g. adding new parameters).

The whole test was relying on a hash parameter order never changing to avoid
passing arguments. I haven't rewritten all of them (only the ones that I
needed), but when I did I changed to passing arguments explicitly with their
fixed value instead.

doudou committed:
fix: adapt the internal ftpd no-overwrite mechanism to the new write+rename scheme
